why-polars-for-data-cleaning/index.qmd

```diff
@@ -0,0 +1,185 @@
+the data is transformed and stored. The checking of the data prior to
+inclusion into Sprout is done within Sprout itself, this work will
+result in a list of errors which the user will need to handle before
+they can load the data into Sprout.
+
+> Which tool should we use for cleaning and processing data before
+> loading it into Sprout?
+
+## Decision drivers
+
+::: content-hidden
+List some reasons for why we need to make this decision and what things
+have arisen that impact work.
+:::
+
+-   We need a tool that can clean data and is easy to integrate with our
+    current workflow.
+-   We want to use an established tool that is well supported.
+-   We want a tool which users will find easy to learn and use.
+-   We do not want to build a tool from scratch.
+
+## Considered options
+
+::: content-hidden
+List and describe some of the options, as well as some of the benefits
+and drawbacks for each option.
+:::
+
+There are a number of tools available to use for data cleaning, but as
+we want something which will run in python, and which is well supported,
+we have narrowed our options down to the following:
+
+-   [DuckDB](https://www.duckdb.org/)
+-   [Pandas](https://pandas.pydata.org/)
+-   [Polars](https://pola.rs/)
+
+### DuckDB
+
+DuckDB is as the name implies not a Python library, but an in-process
+OLAP database solution. It is a very young tool, version 1.0.0 was
+released in 2024, although the first version was released in 2019. The
+developers have set it up as a not-for-profit foundation in the
+Netherlands.
+
+::: columns
+::: column
+#### Benefits
+
+-   Although a very young project it is already in use by a number of
+    large companies and has a large community of users.
+    -   The code is open source, and the way it has been set up is
+        promising for it to remain that way.
+-   It is very fast, and can handle large data sets.
+:::
+
+::: column
+#### Drawbacks
+
+-   It is not designed to be used as a data cleaning tool, but as a
+    database.
+:::
+:::
+
+### Polars
+
+The Polars project started in 2020 and is based in the Netherlands.
+Relatively small team but seems to have a fast growing user base, and
+very enthusiastic developers. Polars is written in Rust and available
+for Python. It is designed to be faster than Pandas, and to handle all
+common data formats. The two main developers are described as
+enthusiastic about open source, which makes it a potentially good choice
+for us.
+
+::: columns
+::: column
+#### Benefits
+
+-   Enthusiastic user community, with a lot of support available.
+-   Supports several languages, including Python.
+-   Handles all common data formats.
+-   Faster than Pandas.
+:::
+
+::: column
+#### Drawbacks
+
+-   Relatively young project, so may not have all the features of more
+    established tools.
+-   Not as well known as Pandas, but certainly getting there.
+-   Compared with Pandas there aren't as many online courses available.
```

Comment:
```suggestion
-   Because it is still new, it isn't as widely used and doesn't have too many learning resources available yet.
```

Avoid comparing to other tools within the benefits and drawbacks. These are about comparisons with our needs.